### PR TITLE
Book details view detail

### PR DESCRIPTION
**Pull request by emmanuelhcpk**
All the features required Done

### CODE REVIEW

File: Project/Book.swift [hunk 1]
```diff
@@ -19,6 +19,9 @@ struct Book {
     var image: String
     var year: String
     var genre: String
+    static public let available = "Available"
+    static public let unavailable = "Unavailable"
```
Comment (acid-wolox) [inline Project/Book.swift:23]: We will surely refactor for this once we get to the localization card in Trello 👍

File: Project/BookDetail/Comments/CommentListViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    private var currentPage : Int  = 1
+
+    public let comments:Property<[CommentViewModel]>
+    public let booksRepository : BooksRepository
+    
+    init(booksRepository : BooksRepository = NetworkingBootstrapper.shared.createBooksRepository() ) {
+        comments = Property(_comments)
+        self.booksRepository = booksRepository
+    }
+    public func loadComments(bookId: Int){
```
Comment (acid-wolox) [inline Project/BookDetail/Comments/CommentListViewModel.swift:26]: Add a space here, please.

File: Project/BookDetail/Comments/CommentListViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        booksRepository.fetchBookComments(bookId: bookId)
+            .map {
+                $0.map {
+                    CommentViewModel(comment: $0)
+                }
+            }
```
Comment (acid-wolox) [inline Project/BookDetail/Comments/CommentListViewModel.swift]: You can go like this here: `.map { $0.map { CommentViewModel(comment: $0) } }`
This way, even if it's kinda less readable, you prioritize the structure of your higher order functions.

File: Project/BookDetail/Comments/CommentListViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                case .success(let commentViewModels) :
+                    self._comments.value += commentViewModels
+                    if(commentViewModels.count > 0){
+                        self.currentPage += 1
+                    }
+                case .failure(let error) :
+                    print(error)
+                }
+        }
+    }
```
Comment (acid-wolox) [inline Project/BookDetail/Comments/CommentListViewModel.swift:41]: Just a minor style adjustment:

```
.startWithResult { [unowned self] result in
    switch result {
    case .success(let commentViewModels) :
        self._comments.value += commentViewModels
        if(commentViewModels.count > 0){
            self.currentPage += 1
        }
    case .failure(let error) :
        print(error)
    }
}
```

File: Project/BookDetail/Comments/CommentListViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    func getByIndex(index:Int) -> CommentViewModel {
+        return comments.value[index]
+    }
+    func getCount() -> Int {
```
Comment (acid-wolox) [inline Project/BookDetail/Comments/CommentListViewModel.swift:47]: And a space here.

File: Project/BookDetail/Comments/CommentViewController.swift [hunk 6]
```diff
@@ -8,13 +8,16 @@
...
+
 class CommentViewController: UIViewController , UITableViewDelegate, UITableViewDataSource{
     
     private var _view: CommentListView = CommentListView.loadFromNib()!
-    
-    init() {
-        super.init(nibName:nil, bundle:nil)
+    private let _viewModel: CommentListViewModel = CommentListViewModel()
+    private var bookId :Int
+    init(bookId: Int) {
```
Comment (acid-wolox) [inline Project/BookDetail/Comments/CommentViewController.swift:19]: Add a space here.

File: Project/BookDetail/Comments/CommentViewController.swift [hunk 7]
```diff
@@ -27,16 +30,46 @@ class CommentViewController: UIViewController , UITableViewDelegate, UITableView
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        _view.commentTable.delegate = self
+        _view.commentTable.rowHeight = 120
+        _view.commentTable.dataSource = self
         _view.commentTable.register(cell: CommentView.self)
-        //view.backgroundColor = .backgroundColor
+        _view.commentTable.separatorStyle = .none
```
Comment (acid-wolox) [inline Project/BookDetail/Comments/CommentViewController.swift:39]: You can send all of this `_view.commentTable` lines to a new function called, I don't know, `func setupCommentTable()`?

File: Project/BookDetail/Comments/CommentViewController.swift [hunk 8]
```diff
@@ -27,16 +30,46 @@ class CommentViewController: UIViewController , UITableViewDelegate, UITableView
...
+
+        _viewModel.loadComments(bookId: bookId)
+        _viewModel.comments.producer.startWithValues { [unowned self] _ in
+            self._view.commentTable.reloadData()
+        }
     }
     //table functions
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return 5
+        print(_viewModel.getCount())
```
Comment (acid-wolox) [inline Project/BookDetail/Comments/CommentViewController.swift]: You can take this out if you're sure that this is working properly.

File: Project/BookDetail/Comments/CommentViewController.swift [hunk 9]
```diff
@@ -27,16 +30,46 @@ class CommentViewController: UIViewController , UITableViewDelegate, UITableView
...
+        return _viewModel.getCount()
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        var cell = tableView.dequeue(cell: BooksCellView.self, for: indexPath)!
+        var cell = tableView.dequeue(cell: CommentView.self, for: indexPath)!
+        let comment = _viewModel.getByIndex(index: indexPath.row)
+        print(comment.content)
```
Comment (acid-wolox) [inline Project/BookDetail/Comments/CommentViewController.swift]: Same as before, you won't need this line in future versions of the code.

File: Project/BookDetail/Comments/CommentViewController.swift [hunk 10]
```diff
@@ -27,16 +30,46 @@ class CommentViewController: UIViewController , UITableViewDelegate, UITableView
...
+                    DispatchQueue.main.async { //to execute on main thread
+                        cell.avatar.image = img
+                        comment.isLoad = true
+                        comment.imageLoad = img
+                    }
+                case let .failure(error):
+                    print("Error Finding Image",error)
+                }
+            }
+        }else{
```
Comment (acid-wolox) [inline Project/BookDetail/Comments/CommentViewController.swift]: Leave a single space between `else` and the curly braces.

File: Project/BookDetail/Detail/DetailsViewController.swift [hunk 11]
```diff
@@ -33,9 +34,22 @@ class DetailsViewController: UIViewController {
         _view.portrait.image = _viewModel.imageLoad
         _view.genre.text = _viewModel.genre
         _view.year.text = _viewModel.year
+        if (_viewModel.status == Book.available){
+            setupAvailableBook()
+        }else{
```
Comment (acid-wolox) [inline Project/BookDetail/Detail/DetailsViewController.swift]: Leave a single space between the `else` and the curly braces.

File: Project/Extensions/UIButtonExtension.swift [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+
+extension UIButton {
+    
+    func enable()  {
+        self.setGradient(colors: [UIColor(hex:"00ADEE")!, UIColor(hex: "38CCCD")!])
+    }
+    
+    /*func disable() {
+        self.setGradient(colors: [UIColor(hex:"EAEAEA")!, UIColor(hex: "F4F4F4")!])
+    }*/
```
Comment (acid-wolox) [inline Project/Extensions/UIButtonExtension.swift]: Erase this if you're not going to implement it.

**Review by acid-wolox — CHANGES_REQUESTED** [review]
Really good work here, Emma. Just a few style corrections and we're done with this PR!

File: Project/Extensions/UIButtonExtension.swift (hunk 12 above)
Comment (emmanuelhcpk) [reply Project/Extensions/UIButtonExtension.swift]: ok sorry , i try to errase every commented function hahaha

**Review by acid-wolox — APPROVED** [review]
Way to go!

**Pull request merged**